### PR TITLE
Enable processing and dynamic registration of new ungridded datasets in AeroVal

### PR DESCRIPTION
**Pull request by jgliss**
See #464

### CODE REVIEW

File: pyaerocom/io/read_mscw_ctm.py [hunk 1]
```diff
@@ -140,12 +141,34 @@ def calc_concNtnh(concnh3,concnh4):
...
+        1000*10,
+        288.15, #15 deg celcius
+        "ug m-3",
+        46.0055, # g/mol NO2
+        to_unit="nmol mol-1"
+    )
+
+    vmrox = vmrno2 + vmro3
+    vmrox.attrs["units"] = "nmol mol-1"
+    return vmrox
```
Comment (dulte) [inline pyaerocom/io/read_mscw_ctm.py:170]: This is not tested by me yet, due to problems with the colocation (see message)

**Comment by jgliss** [comment]
@dulte, even though your EEA evaluation looks still strange I think we should merge this and then further debug from main-dev. I added some tests based on AERONET for the merging (AODtot = AODf + AODc) and this seems to work fine. As discussed, perhaps there is something off with the EEA reader, which we should investigate. Perhaps you could make an issue that adresses the problem you encountered with EEA vmrox=vmrno2+vmro3.

Could you please review the recent changes I did here?

File: pyaerocom/griddeddata.py [hunk 2]
```diff
@@ -2021,16 +2021,19 @@ def aerocom_savename(self, data_id=None, var_name=None,
...
+            if self.metadata['vert_code'] is not None:
+                vert_code=self.metadata['vert_code']
+            else:
+                try:
+                    from pyaerocom.io.fileconventions import FileConventionRead
+                    f = self.from_files[0]
+                    fconv = FileConventionRead().from_file(f)
+                    vert_code = fconv.get_info_from_file(f)['vert_code']
+                except Exception:
+                    pass
```
Comment (dulte) [inline pyaerocom/griddeddata.py:2033]: Maybe write someting to logger before passing?

Comment (jgliss) [reply pyaerocom/griddeddata.py:2033]: Good point. I think this method tries to be too smart, I'll revise in #456

**Review by dulte — APPROVED** [review]
Looks good :)

**Pull request merged**